components/camel-google/camel-google-mail/src/main/java/org/apache/camel/component/google/mail/GoogleMailConfiguration.java

```diff
@@ -124,4 +133,38 @@ public void setApplicationName(String applicationName) {
         this.applicationName = applicationName;
     }
 
+    public String getFilename() {
+        return filename;
+    }
+
+    /**
+     * Filename "*.json" with credentials for Service account
+     */
+    public void setFilename(String filename) {
```

Comment:
This filename option should use the usual approach we have in other component so it should be resolved from classpath, file, http etc. You have multiple examples in the codebase

Reply:
Yes, really.
Replaced "filename" parameter with "keyResource". As example I use "certResource" in [SSH Component](https://camel.apache.org/components/next/ssh-component.html).

Reply:
The keyResource needs to be read through resourceHelper while passing the variable to the makeClient, it should something like https://github.com/apache/camel/blob/main/components/camel-google/camel-google-functions/src/main/java/org/apache/camel/component/google/functions/GoogleCloudFunctionsClientFactory.java#L43

or something like resourceUri in https://camel.apache.org/components/3.14.x/mustache-component.html or any other templating component. I don't know what the makeClient expect, but I suppose this would be a much more complete a flexible approach than passing a string.

Reply:
> what the makeClient expect

makeClient , exactly _com.google.api.client.googleapis.auth.oauth2.GoogleCredential.fromStream(InputStream)_ makes good job with InputStream: he parses it and [**closes** it](https://github.com/googleapis/google-api-java-client/blob/main/google-api-client/src/main/java/com/google/api/client/googleapis/auth/oauth2/GoogleCredential.java#L231). This is the reason, why we transform String into InputStream jst here: in another place we must close/cutch errors self.
I would prefer stay InputStream processing as it is now, but of course if you think it is not a strong argument, I make it the way you proposed

Reply:
I didn't notice the makeClient implementation in the batch client. It is absolutely correct.